docs/ci/ci-automated-release-workflow.md

```diff
@@ -0,0 +1,134 @@
+
+### Goals of the Automation
+
+The automated process should be:
+
+1. Configurable so that any library maintainer can configure the automation to meet their specific repository's needs
+2. Schedulable so that releases are generated according to a specific cadence
+
+Scheduled releases are generally important for setting expectations around release cadences, but they are by no means required for every library and may not even be applicable to some in this GitHub organization.
+
+## Y-Stream Automation Workflow
+
+### Overview of Y-Stream (Minor) Release Automation
+
+Y-stream (minor) releases have historically been handled differently from Z-stream releases. Z-stream releases oftentimes involve backports for bugfixes and may require manual code rebasing to get those backports merged into the appropriate existing release branch. Therefore, we can think of Y-stream release logic as the "basis" for Z-stream release logic, which takes the Y-stream logic and builds upon it to account for backporting and other desirable actions.
+
+### Configurable Components
+
+As mentioned above, there are configurable components within this release process automation. The diagram in the next section references two configurable components:
+
+#### Trigger Schedule
+
+The trigger schedule defines the day and time (in UTC) when the release process will kick off.  This schedule can be disabled if desired, and maintainers can trigger the release process manually when needed instead.
+
+As mentioned in the brief overview of the automation, this in-house `create-automated-release` action will be callable from any workflow file. Thus, each library maintainer who wants to create a scheduled release should first create a  `.github/workflows/automated-release.yml` workflow file in their repository and define a schedule for it. For example:
+
+```yaml
+name: Create Y-Stream Release
+
+on:
+  schedule:
+    - cron: '30 1 1,15 * *' # Triggers at 1:30am UTC every 2 weeks on the 1st and the 15th day of each month
+```
+
+#### Custom List of Dependencies to Cap
+
+With each release, some library maintainers may want to cap the version of certain dependencies within their `requirements.txt` file as well as specify the desired upper cap for each one.
+
+The list of dependencies to cap should be provided in a file called `automated-release-config.yml`. This configuration file may be expanded in the future to accommodate more configurations as needed. Example:
+
+```yaml
+y_stream_release:
+
+  dependency_caps:
+    enable: true # optional parameter. If set to false, then none of the dependencies in `requirements.txt` will be capped.
+    packages:
+      instructlab-sdg: "+0.1.0"
+      instructlab-eval: "+0.2.0"
+      instructlab-training: "+1.0.0"
```

Comment:
Suggestion here, lmk what you think
```suggestion
    packages:
      - name: instructlab-sdg
        greater_than: 0.1.0
      - name: instructlab-eval
        greater_than: 0.2.0
      - name: instructlab-training
        greater_than: 1.0.0
```

You could have similar fields `equals` or `less_than` when applicable